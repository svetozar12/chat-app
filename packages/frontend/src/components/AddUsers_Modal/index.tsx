export { default } from "./AddUsers_Modal";
