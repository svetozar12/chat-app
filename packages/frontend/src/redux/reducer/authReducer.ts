import { Action } from "../state";
import { ActionType } from "../state";
import { InitialState } from "../state";
import { HYDRATE } from "next-redux-wrapper";

const initialState = {
  remember_me: false,
  input: "",
  good: "", //good and bad stand for good alert and bad alert
  bad: "",
  loginPrompt: false,
  cookie: "",
};
const reducer = (state: InitialState = initialState, action: Action) => {
  switch (action.type) {
    case HYDRATE:
      return {
        ...state,
        ...action.payload,
      };
    case ActionType.LOGIN_POST:
      return { ...state, state };
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.good };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.SAVE_INPUT:
      return { ...state, input: action.payload };
    case ActionType.REMEMBER_ME_CHECK:
      return { ...state, remember_me: action.payload };
    case ActionType.QUICK_LOGIN:
      return { ...state, loginPrompt: action.payload };
    case ActionType.SIGN_IN:
      return { ...state, cookie: action.payload };
    case ActionType.SIGN_OUT:
      return { state: undefined };
    default:
      return state;
  }
};

export default reducer;
