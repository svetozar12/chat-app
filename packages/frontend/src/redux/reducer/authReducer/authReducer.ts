import { HYDRATE } from "next-redux-wrapper";
import { IAuthState } from "./state";
import { ActionType } from "../../types";
export const initialState: IAuthState = {
  remember_me: false,
  good: "", //good and bad stand for good alert and bad alert
  bad: "",
  loginPrompt: false,
  cookie: {
    username: "",
    id: "",
    token: "",
    refresh_token: "",
  },
};
const reducer = (state = initialState, action: any) => {
  switch (action.type) {
    case HYDRATE:
      return {
        ...state,
        ...action.payload,
      };
    case ActionType.LOGIN_POST:
      return { ...state, state };
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.good };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.REMEMBER_ME_CHECK:
      return { ...state, remember_me: action.payload };
    case ActionType.QUICK_LOGIN:
      return { ...state, loginPrompt: action.payload };
    case ActionType.SIGN_IN:
      return { ...state, cookie: action.payload };
    default:
      return state;
  }
};

export default reducer;
export { initialState as AuthState };
