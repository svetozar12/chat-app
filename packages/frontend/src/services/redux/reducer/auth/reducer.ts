import { HYDRATE } from 'next-redux-wrapper';
import { IAuth } from './state';
import { ActionType } from '../../types';

export const initialState: IAuth = {
  remember_me: false,
  loginPrompt: false,
};
const authReducer = (state = initialState, action: any) => {
  switch (action.type) {
    case ActionType.REMEMBER_ME_CHECK:
      return { ...state, remember_me: action.payload };
    case ActionType.SIGN_IN:
      return { ...state, cookie: action.payload };
    default:
      return state;
  }
};

export default authReducer;
