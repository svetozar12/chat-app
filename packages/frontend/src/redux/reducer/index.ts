import { combineReducers } from "redux";
import reducer from "./authReducer";
import setReducer from "./setReducer";
import saveInputReducer from "./save_inputReducer";
import messageReducer from "./messageReducer";

const reducers = combineReducers({
  authReducer: reducer,
  setReducer,
  messageReducer,
  saveInputReducer,
});

export default reducers;
