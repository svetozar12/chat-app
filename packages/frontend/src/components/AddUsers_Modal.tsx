import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { InitialState2 } from "../redux/state";
import { GrClose } from "react-icons/gr";
import { CheckBox_component } from "./CheckBox_component";
import { requestUrl } from "../utils/hostUrl_requestUrl";
import { Socket } from "socket.io-client";
import axios from "axios";
export const AddUsers_Modal = ({
  users,
  socketRef,
  setLocalStatus,
  chatId,
  setUsers,
}: {
  users: string[];
  socketRef: Socket;
  setLocalStatus: React.Dispatch<React.SetStateAction<string>>;
  chatId: string;
  setUsers: React.Dispatch<React.SetStateAction<any[]>>;
}) => {
  const [invited, setInvited] = React.useState<string[]>([]);
  const dispatch = useDispatch();
  const state = useSelector(
    (state: { setReducer: InitialState2 }) => state.setReducer,
  );

  const addMembers = async (user: string[]) => {
    try {
      setLocalStatus("d");
      await axios.put(`${requestUrl}/chat-room/${chatId}`, {
        usernames: user,
      });
      setLocalStatus("");
      return true;
    } catch (error) {
      return false;
    }
  };

  const handleSubmit = async () => {
    try {
      if (invited.length <= 0) return;
      await addMembers(invited);
      setUsers(users.filter((element) => !invited.includes(element)));
      socketRef.emit("inviting_multiple_users", { users: invited });
      setInvited([]);

      return true;
    } catch (error) {
      return false;
    }
  };

  return (
    <div className="fRequests_modal">
      <section
        style={{ position: "relative", textAlign: "center" }}
        className="modal_heading flex"
      >
        <h1 style={{ padding: "0 25%" }}>Add people</h1>
        <div
          onClick={() => {
            dispatch({
              type: "SET_MODAL_INVITE",
              payload: !state.setModalInvite,
            });
          }}
          style={{ width: "3rem", height: "3rem" }}
          className="circle_border absolute_top_right flex"
        >
          <GrClose style={{ width: "2rem", height: "2rem" }} />
        </div>
      </section>
      <div
        className="flex"
        style={{ overflowY: "auto", width: "100%", flexDirection: "column" }}
      >
        {users.length === 0 && <h1>No Chat suggestions</h1>}
        {users.map((item, index) => {
          return (
            <CheckBox_component
              key={index}
              invited={invited}
              setInvited={setInvited}
              item={item}
            />
          );
        })}
      </div>
      <div className="modal_footer flex">
        <button onClick={handleSubmit} className="button" type="submit">
          submit
        </button>
      </div>
    </div>
  );
};
