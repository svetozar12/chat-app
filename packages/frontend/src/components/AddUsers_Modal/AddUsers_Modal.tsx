import React from 'react';
import { GrClose } from 'react-icons/gr';
import { css } from '@emotion/css';
import { connect } from 'react-redux';
import { STATE } from 'services/redux/reducer';
import { IWebSocket } from 'services/redux/reducer/websocket/state';
import { IToggle } from 'services/redux/reducer/toggles/state';
import { toggleInviteModal } from 'services/redux/reducer/toggles/actions';
import { bindActionCreators, Dispatch } from 'redux';
import { useCookie } from 'next-cookie';
import { useUpdateChatMutation } from 'services/generated';

interface IAddUserModal {
  users: string[];
  chatId: string;
  ws: IWebSocket;
  toggle: IToggle;
  toggleInviteModal: typeof toggleInviteModal;
  setUsers: React.Dispatch<React.SetStateAction<any[]>>;
}

function AddUserModal(props: IAddUserModal) {
  const { users, chatId, setUsers, toggleInviteModal, ws, toggle } = props;
  const cookie = useCookie();
  const [invited, setInvited] = React.useState<string[]>([]);
  const [updateChat] = useUpdateChatMutation();
  const addMembers = async (user: string[]) => {
    try {
      await updateChat({
        variables: {
          auth: { userId: cookie.get('id'), AccessToken: cookie.get('cookie') },
          chat_id: chatId,
          username: undefined,
          usersData: user,
        },
      });
    } catch (error) {
      return false;
    }
  };

  const handleSubmit = async () => {
    try {
      if (invited.length <= 0) return;
      await addMembers(invited);
      setUsers(users.filter((element) => !invited.includes(element)));
      ws.ws?.emit('inviting_multiple_users', { users: invited });
      setInvited([]);

      return true;
    } catch (error) {
      return false;
    }
  };

  return (
    <div
      className={css`
        position: fixed;
        flex-direction: column;
        display: flex;
        align-items: center;
        justify-content: !important start;
        z-index: 9999;
        width: 40%;
        background: var(--main-white);
        overflow: hidden;
        left: 0;
        right: 0;
        margin-left: auto;
        margin-right: auto;
        box-shadow: 0 0 10px rgba(0, 0, 0, 0.3);
        border-radius: 5px;
        @media (min-width: 1010px) {
          width: 70%;
        }
      `}
    >
      <section style={{ position: 'relative', textAlign: 'center' }} className="modalHeading flex">
        <h1 style={{ padding: '0 25%' }}>Add people</h1>
        <div
          onClick={() => toggleInviteModal(!toggle.toggleInvideModal)}
          style={{ width: '3rem', height: '3rem' }}
          className="circleBorder absoluteTopRight flex"
        >
          <GrClose style={{ width: '2rem', height: '2rem' }} />
        </div>
      </section>
      <div className="flex" style={{ overflowY: 'auto', width: '100%', flexDirection: 'column' }}>
        {users.length === 0 && <h1>No Chat suggestions</h1>}
        {/* {users.map((item, index) => (
          <CheckBox key={index} invited={invited} setInvited={setInvited} item={item} />
        ))} */}
      </div>
      <div className="modalFooter flex">
        <button onClick={handleSubmit} className="button" type="submit">
          submit
        </button>
      </div>
    </div>
  );
}

const mapStateToProps = (state: STATE) => ({
  ws: state.ws,
  toggle: state.toggle,
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  toggleInviteModal: bindActionCreators(toggleInviteModal, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(AddUserModal);
