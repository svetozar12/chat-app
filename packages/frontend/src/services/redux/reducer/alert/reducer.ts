import { AnyAction } from 'redux';
import { IAlert } from 'services/redux/reducer/alert/state';
import { ActionType } from '../../types';

const initialState: IAlert = {
  good: '', // good and bad stand for good alert and bad alert
  bad: '',
};

const alertReducer = (state = initialState, action: AnyAction) => {
  switch (action.type) {
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.payload };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.payload };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.payload };
    default:
      return state;
  }
};

export default alertReducer;
