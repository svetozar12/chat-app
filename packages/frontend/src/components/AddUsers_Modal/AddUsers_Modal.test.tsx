import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { AddUsers_Modal } from "components/AddUsers_Modal/AddUsers_Modal";
import renderer from "react-test-renderer";
import { AuthState } from "redux/reducer/authReducer";
import { render, cleanup, RenderResult } from "@testing-library/react";
import { ReactTestRendererJSON } from "react-test-renderer";
import "@testing-library/jest-dom";

let component: ReactTestRendererJSON | ReactTestRendererJSON[] | null;
let container: RenderResult;
const submit: any = jest.fn();

beforeEach(() => {
  const mockStore = configureStore([]);
  const store = mockStore({
    authReducer: AuthState,
  });

  component = renderer
    .create(
      <Provider store={store}>
        <AddUsers_Modal
          users={["ivan", "greg"]}
          socketRef={submit}
          setLocalStatus={submit}
          setUsers={submit}
          chatId={"321312312321"}
        />
      </Provider>,
    )
    .toJSON();

  container = render(
    <Provider store={store}>
      <AddUsers_Modal
        users={["ivan", "greg"]}
        socketRef={submit}
        setLocalStatus={submit}
        setUsers={submit}
        chatId={"321312312321"}
      />
    </Provider>,
  );
});

afterEach(cleanup);

describe("Render connected React-redux page", () => {
  it("should create snapshot for <AddUsers_Modal/>", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render <AddUsers_Modal/>", () => {
    const renderedComponent = container.getByText("Add people");
    expect(renderedComponent).toBeInTheDocument();
  });
});
