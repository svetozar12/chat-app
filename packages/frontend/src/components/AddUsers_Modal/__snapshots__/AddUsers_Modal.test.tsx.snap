// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render connected React-redux page should create snapshot for <AddUsers_Modal/> 1`] = `
<div
  className="fRequests_modal"
>
  <section
    className="modal_heading flex"
    style={
      Object {
        "position": "relative",
        "textAlign": "center",
      }
    }
  >
    <h1
      style={
        Object {
          "padding": "0 25%",
        }
      }
    >
      Add people
    </h1>
    <div
      className="circle_border absolute_top_right flex"
      onClick={[Function]}
      style={
        Object {
          "height": "3rem",
          "width": "3rem",
        }
      }
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
            "height": "2rem",
            "width": "2rem",
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3,3 L21,21 M3,21 L21,3"
          fill="none"
          stroke="#000"
          strokeWidth="2"
        />
      </svg>
    </div>
  </section>
  <div
    className="flex"
    style={
      Object {
        "flexDirection": "column",
        "overflowY": "auto",
        "width": "100%",
      }
    }
  >
    <div
      className="flex css-23rh6g"
      onClick={[Function]}
    >
      <p
        className="css-1nu6ntq"
      >
        ivan
      </p>
      <input
        checked={false}
        className="css-jgjwtu"
        type="checkbox"
        value="ivan"
      />
    </div>
    <div
      className="flex css-23rh6g"
      onClick={[Function]}
    >
      <p
        className="css-1nu6ntq"
      >
        greg
      </p>
      <input
        checked={false}
        className="css-jgjwtu"
        type="checkbox"
        value="greg"
      />
    </div>
  </div>
  <div
    className="modal_footer flex"
  >
    <button
      className="button"
      onClick={[Function]}
      type="submit"
    >
      submit
    </button>
  </div>
</div>
`;
